src/save-restore.c

```diff
@@ -0,0 +1,403 @@
+/*
+ * Copyright Red Hat
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "config.h"
+
+#include <gio/gio.h>
+
+#include "xdp-context.h"
+#include "xdp-dbus.h"
+#include "xdp-impl-dbus.h"
+#include "xdp-portal-config.h"
+#include "xdp-session.h"
+#include "xdp-utils.h"
+
+#include "save-restore.h"
+
+struct _XdpSaveRestore
+{
+  XdpDbusSaveRestoreSkeleton parent_instance;
+
+  XdpDbusImplSaveRestore *impl;
+  GCancellable *cancellable;
+};
+
+#define XDP_TYPE_SAVE_RESTORE (xdp_save_restore_get_type ())
+G_DECLARE_FINAL_TYPE (XdpSaveRestore,
+                      xdp_save_restore,
+                      XDP, SAVE_RESTORE,
```

Comment:
Hmm so the USB portal names itself XdpUsb, but apparently most every other portal doesn't have an Xdp prefix. And apparently that was intentional

So, IDK am I supposed to rename this portal to just SaveRestore?

Perhaps if we want to maintain a separation, we could have a `Portal` prefix, so that all the portals are `PortalAccounts`, `PortalSaveRestore`, `PortalUsb`, etc and the utility classes/functions can keep `XdpSession`, `XdpContext`, etc